lib/stat_tracker.rb

```diff
@@ -96,30 +80,36 @@ def count_of_teams
     @teams.length
   end
 
-  def average_goals_by_team(team_id, hoa = nil)
-    goals = 0
-    games = 0
+  def average_goals_by_team(team_id, hoa = nil) # game_teams?
+    goals = total_games_and_goals_by_team(team_id, hoa)[0]
```

Comment:
Refactored this method to have the 2 helper methods below.
Used a mock and stub on the test.